app/src/main/kotlin/edu/stanford/bdh/engagehf/MainActivityViewModel.kt

```diff
@@ -36,8 +37,20 @@ class MainActivityViewModel @Inject constructor(
         viewModelScope.launch {
             accountEvents.events.collect { event ->
                 when (event) {
-                    is AccountEvents.Event.SignUpSuccess, AccountEvents.Event.SignInSuccess -> {
-                        navigator.navigateTo(AppNavigationEvent.AppScreen)
+                    is AccountEvents.Event.SignInSuccess, AccountEvents.Event.SignUpSuccess -> {
+                        when (val userState = userSessionManager.getUserState()) {
```

Comment:
minor as we can extract the the common part:

```kotlin
val navigationEvent = when (val userState = userSessionManager.getUserState()) {
    UserState.NotInitialized -> {
        OnboardingNavigationEvent.OnboardingScreen
    }

    is UserState.Registered -> {
        if (userState.hasInvitationCodeConfirmed) {
            AppNavigationEvent.AppScreen
        } else {
            OnboardingNavigationEvent.InvitationCodeScreen
        }
    }
}
navigator.navigateTo(navigationEvent)
```

It would be great to also add a unit test for the `NotInitialized` user state even though that should never be the case actually.